File: tasks/implementations/cpp/tasks/api_client.cpp
#include <fstream>
#include <iostream>
#include <regex>
#include <sstream>
#include <string>

static std::string read_file_api(const std::string& path) {
    std::ifstream file(path);
    std::ostringstream buffer;
    buffer << file.rdbuf();
    return buffer.str();
}

void run_api_client(const std::string& size, const std::string& fixturesRoot) {
    std::string text = read_file_api(fixturesRoot + "/mocks/public_api_" + size + ".json");
    std::regex re(R"re("id":\s*(\d+),\s*"name":\s*"([^"]+)",\s*"value":\s*(\d+))re");
    long long total = 0;
    for (std::sregex_iterator it(text.begin(), text.end(), re), end; it != end; ++it) total += std::stoll((*it)[1].str()) + std::stoll((*it)[3].str()) + (*it)[2].str().size();
    std::cout << total << std::endl;
}

File: tasks/implementations/node/tasks/api_client.js
const fs = require('fs');
module.exports = function(size, fixturesRoot) {
  const text = fs.readFileSync(`${fixturesRoot}/mocks/public_api_${size}.json`, 'utf8');
  let total = 0;
  for (const match of text.matchAll(/"id":\s*(\d+),\s*"name":\s*"([^"]+)",\s*"value":\s*(\d+)/g)) {
    total += Number(match[1]) + Number(match[3]) + match[2].length;
  }
  console.log(total);
};
